html/js/product_edit_form_display.js

```diff
@@ -0,0 +1,188 @@
+/*global lang */
+/* eslint-disable no-undef */
+const nutritionProducts = document.querySelectorAll(".nutrition_product");
```

Comment:
This does not return anything for me.

Did you forgot to add a modification to `templates/web/pages/product_edit/product_edit_form_display.tt.html` ? (to add this element and related data ?).
It's a good idea to add an element with specific properties for each nutrition type, but you have to put it in product_edit_form_display.tt.html

Reply:
Oh yes @alexgarel ..I somehow missed to add the script to `product_edit_form_display.tt.html`. Now it should work fine!